client/txn_test.go

```diff
@@ -368,3 +368,51 @@ func TestRunTransactionRetryOnErrors(t *testing.T) {
 		}
 	}
 }
+
+// TestAbortTransactionOnCommitErrors verifies that non-exec transactions are
+// aborted on the correct errors.
+func TestAbortTransactionOnCommitErrors(t *testing.T) {
+	defer leaktest.AfterTest(t)
+
+	testCases := []struct {
+		err   error
+		abort bool
+	}{
+		{&proto.ReadWithinUncertaintyIntervalError{}, true},
+		{&proto.TransactionAbortedError{}, false},
+		{&proto.TransactionPushError{}, true},
+		{&proto.TransactionRetryError{}, true},
+		{&proto.Error{}, true},
+		{&proto.RangeNotFoundError{}, false},
+		{&proto.RangeKeyMismatchError{}, false},
+		{&proto.TransactionStatusError{}, false},
```

Comment:
Are the above 3 cases correct? They occur due to checking whether the error matches the `proto.TransactionRestartError` interface.